editor.planx.uk/src/pages/FlowEditor/lib/store/settings.ts

```diff
@@ -51,6 +59,55 @@ export const settingsStore: StateCreator<
     return Boolean(result?.id);
   },
 
+  flowDescription: "",
+
+  setFlowDescription: (flowDescription: string) => set({ flowDescription }),
+
+  updateFlowDescription: async (newDescription: string) => {
+    const { id, $client } = get();
+    const result = await $client.flow.setDescription({
+      flow: { id },
+      description: newDescription,
+    });
+    set({ flowDescription: newDescription });
+    return Boolean(result?.id);
+  },
+
+  getFlowSettings: async (flowSlug, teamSlug) => {
```

Comment:
I found the naming here is a little unclear / unexpected - we already have a `flowSettings` property and this is not a getter for that property, but rather a function which fetches settings, status and description of a flow.

I think I'd stick to the existing pattern of -
 - Fetch within the route
 - Set store variables
 - Child components read from stores
 
 Always using the route `req` object over store values also ensures we're always fetching the right data and there's no room for any race condition or unexpected behaviour. This relies on a user always navigating via a route which sets these.

Reply:
https://github.com/user-attachments/assets/001fee03-b878-4deb-a82d-1f00b24631f1

Reply:
Yeah good thinking, I did keep the function name the same for consistency, but probably need some altered naming here. It is complicated a bit by the current `flowSettings` not really being all the settings for a flow, and rather an object of footer, privacy, and help rich text info. I'll look to update the naming to make it clearer 

The `req` bit is a great catch, sorry for missing this. Added the change now. 

Would there be a reason why these settings maintain a different pattern to General Settings and Design Settings where we fetch within the index.tsx of the component? 

Think I've fallen into a trap of trying to partially improve code while trying to debug and creating wee errors, thanks for all the help and time

Reply:
> Would there be a reason why these settings maintain a different pattern to General Settings and Design Settings where we fetch within the index.tsx of the component?

Honestly, I think it's just history! This has been around for a while and we've not yet taken the dive to standardise and update it to match the rest - not a pressing concern right now, but would be a super nice tidy up in future.

> Think I've fallen into a trap of trying to partially improve code while trying to debug and creating wee errors, thanks for all the help and time

Very easily done - especially in sections of the code like this which are too low on test coverage.